src/cartservice/cartstore/InstrumentedCartStore.cs

```diff
@@ -0,0 +1,116 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Threading.Tasks;
+using cartservice.interfaces;
+using Hipstershop;
+using OpenCensus.Trace;
+using OpenCensus.Trace.Sampler;
+
+namespace cartservice.cartstore
+{
+    /// <summary>
+    /// Wrapper for Cart Store - instrumented with OpenCensus tracing
+    /// </summary>
+    internal class InstrumentedCartStore : ICartStore
```

Comment:
Why do we have a wrapper? Can't we just instrument the code? 

Is this a pattern in C#?

Reply:
Spans should be named after the logical operation they represent and not after a method name. Method names are implementation detail that can change as a result of refactoring. Span names should never change as it will bust correlation between current and historical data about the same operation.